src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -140,6 +145,183 @@ class_implements($class_name)
   /**
    * {@inheritdoc}
    */
+  public function value(DataSourceInterface $source) {
+    // Return if this field is a callback.
```

Comment:
> The ideal outcome is that we return the callback.

I think it would be best to throw an error with a good explanation, to educate the poor developer ;)